code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePurge.java

```diff
@@ -27,12 +40,29 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
             deactivateThenRemove(link, state.getData());
         }
 
-        for (Mount mount : getObjectManager().children(instance, Mount.class)) {
-            deactivateThenRemove(mount, state.getData());
-        }
+        deleteVolumes(instance, state);
 
         deallocate(instance, null);
 
         return null;
     }
+
+    private void deleteVolumes(Instance instance, ProcessState state) {
+        Object b = DataAccessor.fieldMap(instance, FIELD_LABELS).get(SystemLabels.LABEL_VOLUME_CLEANUP_STRATEGY);
+        String behavior = b != null ? b.toString() : VOLUME_CLEANUP_STRATEGY_UNNAMED;
```

Comment:
Random comment, not needed to be done: We need a util method for get label that returns a string.  I have this same code in a bunch of places.